File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/JMath.java
package com.wsr.gpu;

public class JMath {
    public native void exp(long x, long result, long context);
    public native void ln(long x, float e, long result, long context);
    public native void sigmoid(long x, long result, long context);
    public native void pow(long x, int n, long result, long context);
    public native void sqrt(long x, float e, long result, long context);
}

File: buffer/cpu/src/jvmAndroidMain/kotlin/com/wsr/knist/cpu/elementwise/math/JMath.kt
package com.wsr.knist.cpu.elementwise.math


object JMath {
    external fun exp(x: Long, result: Long)
    external fun ln(x: Long, e: Float, result: Long)
    external fun sigmoid(x: Long, result: Long)
    external fun pow(x: Long, n: Int, result: Long)
    external fun sqrt(x: Long, e: Float, result: Long)
}
